### ISSUE TITLE
libmamba: Cannot find a valid extracted directory cache for 'cuda-nvvp-11.7.50-0.tar.bz2

### ISSUE THREAD

**Issue by addytheyoung** [oobabooga/text-generation-webui#834]
### Describe the bug

Windows 11 for the 1 click install.bat. What's going on here? I've tried with cuda both installed and uninstalled by default. 

EDIT: I have also tried with no spaces in any part of the pathname, same error. 
EDIT2: 'cuda-nvvp-11.7.50.tar.bz2' is in the mamba pkgs in installer_files. What the hell?

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

I have tried running mutliple times, tried with this version of cuda installed and uninstalled before installation, and searched for existing issues.

### Screenshot

![Screenshot 2023-04-06 105655](https://user-images.githubusercontent.com/40842550/230458346-01b222b9-7b7d-43e8-b662-ea94df925698.png)

### Logs

```shell
warning  libmamba [cuda-cuobjdump-11.7.50-0] The following files were already present in the environment:
    - LICENSE
    - build_env_setup.bat
    - conda_build.bat
    - metadata_conda_debug.yaml
Linking cuda-nvprune-11.7.50-0
warning  libmamba [cuda-nvprune-11.7.50-0] The following files were already present in the environment:
    - LICENSE
    - build_env_setup.bat
    - conda_build.bat
    - metadata_conda_debug.yaml
Linking cuda-cccl-11.7.58-0
Linking cuda-nsight-compute-11.7.0-0
Linking libcusparse-dev-11.7.3.50-0
Linking libnpp-dev-11.7.3.21-0
Linking libcusolver-dev-11.3.5.50-0
Linking libcufft-dev-10.7.2.50-0
Linking cuda-nvrtc-dev-11.7.50-0
Linking libcurand-dev-10.2.10.50-0
Linking libnvjpeg-dev-11.7.2.34-0
Linking libcublas-dev-11.10.1.25-0
Linking cuda-libraries-11.7.0-0
Linking cuda-nvvp-11.7.50-0
warning  libmamba Invalid package cache, file 'C:\Users\Andrew\Downloads\oobabooga-windows (1)\oobabooga-windows\instalg
error    libmamba Cannot find a valid extracted directory cache for 'cuda-nvvp-11.7.50-0.tar.bz2'
critical libmamba Package cache error.
```

### System Info

```shell
Windows 11, NVIDIA 3090.
```

**Comment by addytheyoung** [comment]
Update: It looks like it is installed, so what's the error referring to? 

![Screenshot 2023-04-06 113405](https://user-images.githubusercontent.com/40842550/230465620-05740428-a029-4e2d-97a5-4004a5e932dc.png)

**Comment by eevee-k** [comment]
same issue but on Windows 10, with NVIDIA 4090
![cmd_QXKxe5CCdA](https://user-images.githubusercontent.com/130095732/230472410-b9fd13d4-0645-49e2-b321-81d60ab7b1cf.png)

**Comment by Adi256** [comment]
Same issue on Windows 11 Nvidia RTX3070 Laptop GPU

**Comment by Myhamof** [comment]
Same here Win 11 RTX 2080 Super

**Comment by Vagrin** [comment]
Ditto. Windows 10 RTX3060

**Comment by blvckclxud** [comment]
same, windows 10 rtx 3060ti.

**Comment by cmschueths** [comment]
I get a similar error, except it happens when libmamba installs git-2.40.0-h57928b3_1:

![image](https://user-images.githubusercontent.com/35963007/230531120-75a216e0-668c-4189-b172-77b0014b3e6c.png)

No warnings or errors above this. Happens consistently. Trying to re-run the installer afterwards results in `Conda environment is empty.` until I delete installer_files folder and try again, at which point it fails at the same place.

Windows 11, RTX 3090.

**Comment by euri412** [comment]
Same here, windows 10, RTX 3060, 12gb

**Comment by FrameEnder** [comment]
I'm having the same issue with my Windows 11, RTX 3060 12gb GPU.

To be specific I attempted to Install this onto a drive that wasn't my C:/ Drive. not sure if the issue pertains to permissions management, as I had an issue with installing Automatic 1111 through git on my external HDD too that had to do with dubious ownership

**Comment by Vagrin** [comment]
> 

Excellent point. I was trying to install to my D drive. (Very little room on C.) Maybe this is the issue?

**Comment by FrameEnder** [comment]
> > 
> 
> Excellent point. I was trying to install to my D drive. (Very little room on C.) Maybe this is the issue?

Ya, I have the same issue. although I did just barely have enough storage to test it, and I can confirm I got past this issue by just installing on the C:/ Drive Root, I don't know what's holding it back though, but the issue seems to be related to External Drives in some way. I don't think moving the files to an External Drive will entirely work, but I'll give it a try, and get back with you on whether, or not it's an option until whatever is causing this issue is fully sorted out.

**Comment by eevee-k** [comment]
I was installing it to C drive when i got the issue 🤷

**Comment by FrameEnder** [comment]
> I was installing it to C drive when i got the issue 🤷

What is your drive partioned as exFAT, or NTFS

**Comment by Schakuun** [comment]
> > > 
> > 
> > 
> > Excellent point. I was trying to install to my D drive. (Very little room on C.) Maybe this is the issue?
> 
> Ya, I have the same issue. although I did just barely have enough storage to test it, and I can confirm I got past this issue by just installing on the C:/ Drive Root, I don't know what's holding it back though, but the issue seems to be related to External Drives in some way. I don't think moving the files to an External Drive will entirely work, but I'll give it a try, and get back with you on whether, or not it's an option until whatever is causing this issue is fully sorted out.

For me it work with a external Drive, thanks for the Solution, hopefully this will get fixed that it can be used in the C: Drive

**Comment by FrameEnder** [comment]
I attempted to transfer the C Drive version to my other drive, and to sum it up, the process failed. I expected this as enviroments are fairly fragile to this kind of approach, although I also tried something else at the same time.

I just installed it to another Internal Drive, a secondary M.2 that I just bought that was formatted to NTFS, and that leads me to guess it could have something to do with either the file system, as I have only succeeded in installing this to NTFS drives so far, while exFAT drives seem to fail, but as eevee-k has seemed to fail on the C drive which should be NTFS, then other probability is this is somehow related to drive Ownership/Security settings, and/or reading/writing paths for python enviroments, something I know little about, but seems to make sense. I don't know how to fix python issues like that myself, but that might be something worth considering

**Comment by Vagrin** [comment]
Thanks for getting back on that. From other comments it doesn't seem like C or other drives is the issue and possibly not even file format. I think your idea about security/permissions might be the best line of inquiry to follow now. Can you tell me how many gigs it ends up being? It will give me an idea of how wise it will be to set up on my C drive just to play around with it for a while.

**Comment by Vagrin** [comment]
Crap. It does work on drive C: for me but if I download models for it I'll have serious trouble or even just run out of room. So... I do hope they find a way to fix this. There was quite a long time there where PCs came with a small SSD for C: drive (now much larger ones are reasonably priced) and a big disk drive for storage. A lot of people are going to get stuck.

**Comment by FrameEnder** [comment]
> Crap. It does work on drive C: for me but if I download models for it I'll have serious trouble or even just run out of room. So... I do hope they find a way to fix this. There was quite a long time there where PCs came with a small SSD for C: drive (now much larger ones are reasonably priced) and a big disk drive for storage. A lot of people are going to get stuck.

I agree, 31GB's, (the final size for the initial install) might not seem like much to some, but the price for a larger drive isn't cheap, and some still have their PC's boot drive configured with smaller space to save some initial money. that's especially given the cost of computer components in-general are so high, even more so when your building High End rigs that can handle AI tasks. While I understand this, it might be worth considering if your PC can at all handle it, to consider saving money, and buying an inexpensive SSD from Walmart, they do sell 500gb, and even 1tb M.2's (specifically Western Digital branded), I've noticed the one I bought from Walmart seems to work for this use case, and at least in my Area the 1tb SSD came out at about $95 USD the 500gb one was half the price, just food for thought.

**Comment by Myhamof** [comment]
You might want to use symlink then, it's a special shortcut that allows you to move the folder to any drive even a network one but make it seems like it still there, I haven't tried using symlink for the whole program yet (I just move the models folder to my NAS) all you have to do is to move the folder wherever you want it to be and then in the folder where it used to be right click and choose open in Terminal, write "mklink /d name_of_folder path_to_new_location" example "mklink /d models z:\textgen\models" make sure to give the folder the same name as the folder you moved also note that based on your user permission you may have to run this as an administrator, after that, you will have the real folder in another drive whiles appeared to be also in your C drive.

**Comment by Vagrin** [comment]
> > Crap. It does work on drive C: for me but if I download models for it I'll have serious trouble or even just run out of room. So... I do hope they find a way to fix this. There was quite a long time there where PCs came with a small SSD for C: drive (now much larger ones are reasonably priced) and a big disk drive for storage. A lot of people are going to get stuck.
> 
> I agree, 31GB's, (the final size for the initial install) might not seem like much to some, but the price for a larger drive isn't cheap, and some still have their PC's boot drive configured with smaller space to save some initial money. that's especially given the cost of computer components in-general are so high, even more so when your building High End rigs that can handle AI tasks. While I understand this, it might be worth considering that is if your PC can handle a second internal drive, to definately consider saving money, and buying an inexpensive SSD from Walmart, they do sell 500gb, and even 1tb M.2's (specifically Western Digital branded), I've noticed the one I bought from Walmart seems to work for this use case, and at least in my Area the 1tb SSD came out at about $95 USD, and the 500gb one was half the price, just food for thought.

One terabyte would definitely solve that problem for the life the machine. It's been so long since I've cloned a C drive. I have a phobia about it! Always have. The idea of losing everything and trying to start from scratch. And in the early days cloning wasn't 100% effective. I'll probably have to steel myself and just do it. Thanks!

**Comment by Vagrin** [comment]
> You might want to use symlink then, it's a special shortcut that allows you to move the folder to any drive even a network one but make it seems like it still there, I haven't tried using symlink for the whole program yet (I just move the models folder to my NAS) all you have to do is to move the folder wherever you want it to be and then in the folder where it used to be right click and choose open in Terminal, write "mklink /d name_of_folder path_to_new_location" example "mklink /d models z:\textgen\models" make sure to give the folder the same name as the folder you moved also note that based on your user permission you may have to run this as an administrator, after that, you will have the real folder in another drive whiles appeared to be also in your C drive.

Is that built into Windows 10? I thought I'd have to install a program. I'd want to move the two folders install creates as well as the models. I'm operating with a 250G C: drive folks! Never been a problem until I had to install python and other stuff on C to experiment/play with all of our new AI toys. It's a great option. Thank you!

**Comment by Myhamof** [comment]
Yes, it is built into Windows, in the past, it was way less powerful and you would have to download a program to make it more useful but ever since Windows 10 it improved a lot, I use it to install games, programs, and my a.i models to my network drive so I only need to install this once and can access it from all of my devices and I don't need to waste space on my main device

**Comment by Vagrin** [comment]
I just ran into a new problem which might be connected to our original one. I installed it on C: and then tried copying it to my D: drive so I could symlink it as Myhamof suggested and it errored out when it was almost done "Destination Path Too Long" com.nvidia.cuda.ide.editor.feature was what it quit on. If you look in those directories there are thousands of files. If they just clean it all up a bit it might start working on most people's PCs regardless of drive selection. And it might be responsible for the original installation failure.

**Comment by Myhamof** [comment]
> I just ran into a new problem which might be connected to our original one. I installed it on C: and then tried copying it to my D: drive so I could symlink it as Myhamof suggested and it errored out when it was almost done "Destination Path Too Long" com.nvidia.cuda.ide.editor.feature was what it quit on. If you look in those directories there are thousands of files. If they just clean it all up a bit it might start working on most people's PCs regardless of drive selection. And it might be responsible for the original installation failure.

This is a limitation of the Win32 API which limit the path to 260 alot of python packages have a really long name, you can fix it by enabling long path or maybe try to move it to the root of your d drive to short the path, if I remember correctly while installing python there should be a option in the installer to disable the path limit

**Comment by Vagrin** [comment]
> This is a limitation of the Win32 API which limit the path to 260 alot of python packages have a really long name, you can fix it by enabling long path or maybe try to move it to the root of your d drive to short the path, if I remember correctly while installing python there should be a option in the installer to disable the path limit

Thanks friend. I just hit skip three times so I guess it was just a few. It works... like crap. But I'm guessing that's just the stage we're at now. I figured, there's a lot of stuff in here, maybe not all of it is necessary. For doing things I'm not going to do. I'd rarely operate that way but it was worth a try. The symlink worked perfectly. Thanks very much for that idea. I'll look into enabling long path!

**Comment by Myhamof** [comment]
I'm glad it helped you, yea this is still early to run text A.I locally but I hope that open sourcing it will hopefully make it much better and way more accessible just like Stable diffusion
